packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -114,7 +115,16 @@ class ConfigChainBuilder {
       options = jsonCache[content] = jsonCache[content] || json.parse(content);
       if (key) options = options[key];
     } catch (err) {
-      err.message = `${loc}: Error while parsing JSON - ${err.message}`;
+      const code = codeFrame(content, err.lineNumber, err.columnNumber);
```

Comment:
We should check `typeof err.lineNumber === "number"` when generating the code frame and skip it otherwise, I think, since `.lineNumber` isn't a speced property.